From 0bc7317a0dc1045ab22cedf840891c6dba83832b Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Mon, 20 Jun 2016 13:21:52 +0300
Subject: [PATCH 0574/2241] fix: sar: ap806: update sample at reset CPU
 frequency mode

Sync the table with the HW-SPEC tables

Change-Id: I2830304740d4ac1530c2c2c1c95a56631a24340b
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/30567
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Reviewed-by: Neta Zur Hershkovits <neta@marvell.com>
---
 drivers/clk/mvebu/ap806-system-controller.c | 26 +++++++++++++++++---------
 1 file changed, 17 insertions(+), 9 deletions(-)

diff --git a/drivers/clk/mvebu/ap806-system-controller.c b/drivers/clk/mvebu/ap806-system-controller.c
index 3235b37..7d9e892 100644
--- a/drivers/clk/mvebu/ap806-system-controller.c
+++ b/drivers/clk/mvebu/ap806-system-controller.c
@@ -53,21 +53,29 @@ static void __init ap806_syscon_clk_init(struct device_node *np)
 
 	freq_mode = reg & AP806_SAR_CLKFREQ_MODE_MASK;
 	switch (freq_mode) {
-	case 0x0 ... 0x5:
-		cpuclk_freq = 2000;
-		break;
-	case 0x6 ... 0xB:
-		cpuclk_freq = 1800;
-		break;
-	case 0xC ... 0x11:
+	case 0x4:
+	case 0xB ... 0x12:
 		cpuclk_freq = 1600;
 		break;
-	case 0x12 ... 0x16:
+	case 0x1A:
 		cpuclk_freq = 1400;
 		break;
-	case 0x17 ... 0x19:
+	case 0x14 ... 0x18:
 		cpuclk_freq = 1300;
 		break;
+	case 0x19:
+		cpuclk_freq = 1200;
+		break;
+	case 0x13:
+	case 0x1D:
+		cpuclk_freq = 1000;
+		break;
+	case 0x1C:
+		cpuclk_freq = 800;
+		break;
+	case 0x1B:
+		cpuclk_freq = 600;
+		break;
 	default:
 		/* set cpuclk_freq as invalid value to continue and
 		** configure the MSS clock (used to calculate the
-- 
2.7.4

